src/ameba/rule/performance/chained_calls_with_no_bang.cr

```diff
@@ -0,0 +1,69 @@
+module Ameba::Rule::Performance
+  # This rule is used to identify usage of chained calls not utilizing
+  # the bang method variants.
+  #
+  # For example, this is considered inefficient:
+  #
+  # ```
+  # names = %w[Alice Bob]
+  # chars = names
+  #   .flat_map(&.chars)
+  #   .uniq
+  #   .sort
+  # ```
+  #
+  # And can be written as this:
+  #
+  # ```
+  # names = %w[Alice Bob]
+  # chars = names
+  #   .flat_map(&.chars)
+  #   .uniq!
+  #   .sort!
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Performance/ChainedCallsWithNoBang
+  #   Enabled: true
+  # ```
+  class ChainedCallsWithNoBang < Base
+    properties do
+      description "Identifies usage of chained calls not utilizing the bang method variants."
+    end
+
+    # All of those have bang method variants returning `self`
+    # and are not modifying the reveiver type (like `compact` does),
+    # thus are safe to switch to the bang variant.
+    CALL_NAMES = %w(uniq sort sort_by shuffle reverse rotate)
```

Comment:
i would make a configurable property to control this list. Example:

https://github.com/crystal-ameba/ameba/blob/7aa7efd4bdf08bfefb1901c97fe6db13ea1aafdf/src/ameba/rule/performance/size_after_filter.cr#L29-L31